librz/il/il_opcodes.c

```diff
@@ -205,6 +232,26 @@ RZ_API RZ_OWN RzILOpBool *rz_il_op_new_bitv_from_st64(ut32 length, st64 number)
 	return ret;
 }
 
+/**
+ * \brief Creates a new bitvector from the bits \p n to \p m of src bitvector \p v ('s bitv -> 'a bitv).
+ *
+ * \param v Src bitvector
+ * \param n Start index (counted from LSB).
+ * \param m End index (counted from LSB).
+ * \return Bitvector with bits m:n from \p v.
+ */
+RZ_API RZ_OWN RzILOpBool *rz_il_op_new_bitv_from_bitv(RZ_NONNULL RZ_BORROW RzILOpBitVector *v, ut32 m, ut32 n) {
+	rz_return_val_if_fail(v, NULL);
+	rz_return_val_if_fail(m <= n, NULL);
+	RzILOpBitVector *res = rz_il_op_new_bitv_from_ut64(m - n, 0);
+	if (!res) {
+		rz_warn_if_reached();
+		return NULL;
+	}
+	rz_bv_copy_nbits(PURE_BV_GET(v), n, PURE_BV_GET(res), 0, m - n);
```

Comment:
This is wrong, it is confusing emulation with op construction. Essentially, you will want to write something like:
```c
RZ_API RZ_OWN RzILOpBitVector *rz_il_op_new_extract(ut32 hi, ut32 lo, RzILOpBitVector *x) {
	if (lo) {
		x = rz_il_op_new_shiftr(rz_il_op_new_b0(), x, rz_il_op_new_bitv_from_ut64(32, lo));
	}
	return rz_il_op_new_unsigned(hi - lo, x);
}
```

But I think you should reconsider if you really need this because the bit count of `x` is not known without doing full type checking, which is impossible here. So the `32` here is rather arbitrary (but still works), rather than the somewhat cleaner log2(bit count of x).
This is why I never introduced `extract` while doing arm and 6502 and instead always wrote shifts and casts manually.